DatadogInternal/Sources/MessageBus/FeatureMessage.swift

```diff
@@ -39,3 +26,30 @@ public enum FeatureMessage {
     /// The core can send telemtry data coming from all Features.
     case telemetry(TelemetryMessage)
 }
+
+extension FeatureMessage {
+    /// Creates a `.baggage` message with thegiven label and `Encodable` type.
+    ///
+    /// - Parameters:
+    ///   - key: The baggage key.
+    ///   - baggage: The baggage value.
+    /// - Returns: a `.baggage` case.
+    public static func baggage<Value>(key: String, value: Value) -> FeatureMessage where Value: Encodable {
+        .baggage(key: key, baggage: .init(value))
+    }
+
+    /// Decodes the value of a baggage if the label matches.
+    ///
+    /// - Parameters:
+    ///   - label: The requested baggage label.
+    ///   - type: The expected type of the baggage value.
+    /// - Returns: The decoded baggage value, or nil if the label doesn't match.
+    /// - Throws: A `DecodingError` if decoding fails.
+    public func baggage<Value>(forKey key: String, type: Value.Type = Value.self) throws -> Value? where Value: Decodable {
+        guard case let .baggage(messageKey, baggage) = self, messageKey == key else {
+            return nil
```

Comment:
question/ What does it mean when the program reaches this `else { return nil }`? 
* Is it expected to end up here, hence we decide to ignore it by `return nil`?
* Or is it unexpected, but we return `nil` with lack of other options?

I wonder if we shouldn't make it strict by throwing an error, or assert it with telemetry.

Reply:
`nil` is expected here, this is a convenient method that message receivers can use to check for a specific baggage message. Maybe it doesn't read well 🤔 I could be like the following:
```swift
/// Returns the baggage if the key matches the message.
///
/// - Parameters:
///   - key: The requested baggage key.
public func baggage(forKey key: String) -> NewFeatureBaggage? {
    guard case let .baggage(messageKey, baggage) = self, messageKey == key else {
        return nil
    }

    return baggage
}
```

Then a receiver will have to do:
```swift
func receive(message: FeatureMessage, from core: DatadogCoreProtocol) -> Bool {
    do {
        guard let baggage = try message.baggage(forKey: MessageKeys.crash) else {
            return false
        }

        let crash: Crash = try bagagge.decode()
        return send(report: crash.report, with: crash.context, to: core)
    } catch {
        core.telemetry
            .error("Fails to decode crash from RUM", error: error)
    }

    return false
}
```

This might read better, WDYT?

Reply:
Thanks, I can see it now. So the `nil` is unavoidable, but it is rather a cost of `self` being an `enum`, rather than a flaw in the logic we add. Let's keep what we have, this is enough convenience 👌.